File: G_Projects/G_Inventory_Management_System/ProductFactory/ConcreteProducts/GroceryProduct.java
package G_Projects.G_Inventory_Management_System.ProductFactory.ConcreteProducts;

import G_Projects.G_Inventory_Management_System.CommonEnums.ProductEnum;
import G_Projects.G_Inventory_Management_System.ProductFactory.Product;
import java.util.Date;

public class GroceryProduct extends Product{
    private Date expiryDate;
    private boolean refrigerated;

    public GroceryProduct(String id, String name, double price, int quantity, int threshold){
        setId(id);
        setName(name);
        setPrice(price);
        setQuantity(quantity);
        setThreshold(threshold);
        setCategory(ProductEnum.GROCERY);
    }

    public Date getExpiryDate() {
        return expiryDate;
    }
    public void setExpiryDate(Date expiryDate) {
        this.expiryDate = expiryDate;
    }

    public boolean isRefrigerated() {
        return refrigerated;
    }
    public void setRefrigerated(boolean refrigerated) {
        this.refrigerated = refrigerated;
    }
}

File: Python/A_Project/D_Inventory_Management_System/ProductFactory/ConcreteProducts/grocery_product.py
from datetime import date

from CommonEnums.product_enum import ProductEnum
from ProductFactory.product import Product


class GroceryProduct(Product):
    def __init__(self, product_id, name, price, quantity, threshold):
        super().__init__()
        self._expiry_date = None
        self._refrigerated = False
        self.set_id(product_id)
        self.set_name(name)
        self.set_price(price)
        self.set_quantity(quantity)
        self.set_threshold(threshold)
        self.set_category(ProductEnum.GROCERY)

    def get_expiry_date(self):
        return self._expiry_date

    def set_expiry_date(self, expiry_date: date):
        self._expiry_date = expiry_date

    def is_refrigerated(self):
        return self._refrigerated

    def set_refrigerated(self, refrigerated: bool):
        self._refrigerated = refrigerated
